src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyNonCacheableResponse.java

```diff
@@ -77,186 +82,116 @@ public void onResponseMessage(
         switch (msgTypeId)
         {
             case BeginFW.TYPE_ID:
-                final BeginFW begin = this.streamFactory.beginRO.wrap(buffer, index, index + length);
-                handleBegin(begin);
+                final BeginFW begin = httpCacheProxyFactory.beginRO.wrap(buffer, index, index + length);
+                onBegin(begin);
+                break;
+            case DataFW.TYPE_ID:
+                final DataFW data = httpCacheProxyFactory.dataRO.wrap(buffer, index, index + length);
+                onData(data);
+                break;
+            case EndFW.TYPE_ID:
+                final EndFW end = httpCacheProxyFactory.endRO.wrap(buffer, index, index + length);
+                onEnd(end);
+                break;
+            case AbortFW.TYPE_ID:
+                final AbortFW abort = httpCacheProxyFactory.abortRO.wrap(buffer, index, index + length);
+                onAbort(abort);
                 break;
             case WindowFW.TYPE_ID:
-                final WindowFW window = streamFactory.windowRO.wrap(buffer, index, index + length);
-                onWindowWhenProxying(window);
+                final WindowFW window = httpCacheProxyFactory.windowRO.wrap(buffer, index, index + length);
+                onWindow(window);
                 break;
             case ResetFW.TYPE_ID:
-                final ResetFW reset = streamFactory.resetRO.wrap(buffer, index, index + length);
-                onResetWhenProxying(reset);
-                break;
-            default:
-                // ignore
+                final ResetFW reset = httpCacheProxyFactory.resetRO.wrap(buffer, index, index + length);
+                onReset(reset);
                 break;
         }
     }
 
-    void handleStream(
-        int msgTypeId,
-        DirectBuffer buffer,
-        int index,
-        int length)
-    {
-        streamState.accept(msgTypeId, buffer, index, length);
-    }
-
-    @Override
-    public String toString()
-    {
-        return String.format("%s[connectRouteId=%016x, connectReplyStreamId=%d, acceptReplyBudget=%016x, " +
-                " padding=%d]", getClass().getSimpleName(),
-            connectRouteId, connectReplyStreamId, acceptReplyBudget, padding);
-    }
-
-
-    private void handleBegin(
+    private void onBegin(
         BeginFW begin)
     {
         final long connectReplyId = begin.streamId();
-        traceId = begin.trace();
+        final long traceId = begin.trace();
 
-        final OctetsFW extension = streamFactory.beginRO.extension();
+        final OctetsFW extension = httpCacheProxyFactory.beginRO.extension();
 
         if (extension.sizeof() > 0)
         {
-            final HttpBeginExFW httpBeginFW = extension.get(streamFactory.httpBeginExRO::wrap);
+            final HttpBeginExFW httpBeginFW = extension.get(httpCacheProxyFactory.httpBeginExRO::wrap);
```

Comment:
Use `tryWrap` and test for non-null `httpBeginEx` instead of checking for empty extension?

Reply:
Note `httpBeginFW` should be `httpBeginEx`